Jenkinsfile

```diff
@@ -54,23 +54,98 @@ def doc_only_change() {
 }
 
 def skip_stage(String stage) {
-    return commitPragma(pragma: 'Skip-' + stage).contains('true')
+    return cachedCommitPragma(pragma: 'Skip-' + stage).contains('true')
 }
 
 def quickbuild() {
-    return commitPragma(pragma: 'Quick-build') == 'true'
+    return cachedCommitPragma(pragma: 'Quick-build') == 'true'
+}
+
+def functional_post_always() {
+   return sh(label: "Job Cleanup",
+             script: '''rm -rf install/lib/daos/TESTING/ftest/avocado/job-results/job-*/html/
+                        # Remove the latest avocado symlink directory to avoid inclusion in the
+                        # jenkins build artifacts
+                        unlink install/lib/daos/TESTING/ftest/avocado/job-results/latest
+                        rm -rf "Functional/"
+                        mkdir "Functional/"
+                        # compress those potentially huge DAOS logs
+                        if daos_logs=$(find install/lib/daos/TESTING/ftest/avocado/job-results/job-*/daos_logs/* -maxdepth 0 -type f -size +1M); then
+                            lbzip2 $daos_logs
+                        fi
+                        arts="$arts$(ls *daos{,_agent}.log* 2>/dev/null)" && arts="$arts"$'\n'
+                        arts="$arts$(ls -d install/lib/daos/TESTING/ftest/avocado/job-results/job-* 2>/dev/null)" && arts="$arts"$'\n'
+                        if [ -n "$arts" ]; then
+                            mv $(echo $arts | tr '\n' ' ') "Functional/"
+                        fi''',
```

Comment:
Can this script be moved into something like a "jenkins" or "util/jenkins" directory?

Reply:
Likely, but that seems like it's more suited to refactor work, and how it will generally handle scriptlets, which I think needs to be thought out and designed a bit before we start on it.

In general I think our use of scriptets inside of scriptlets (i.e. in `{s,clu}sh` commands) needs some thought on how they can be better handled so as to not need all of the manual escaping that they do.  I have an idea or two on that that I need to test and flesh our.